src/ui/Base/RegisteredNamedMethods.ts

```diff
@@ -248,17 +248,26 @@ function getCoveoAnalytics(element: HTMLElement) {
  * service automatically converts white spaces to underscores, and uppercase characters to lowercase characters in key
  * names. Each value must be a simple string. If you do not need to log metadata, you can simply pass an empty JSON
  * ( `{}` ).
+ * @param result The IQueryResult that the custom event is linked to, if any.
  */
-export function logCustomEvent(element: HTMLElement, customEventCause: IAnalyticsActionCause, metadata: IStringMap<string>) {
+export function logCustomEvent(
+  element: HTMLElement,
+  customEventCause: IAnalyticsActionCause,
+  metadata: IStringMap<string>,
+  result: IQueryResult
```

Comment:
I think for backward compatibility, `result` should be an optional parameter. 
It was also this case in the previous PR https://github.com/coveo/search-ui/pull/923

Reply:
Yep, exactly !

Reply:
My bad